### PR TITLE
add input data type property and filter

### PR DESCRIPTION
**Pull request by rrchai**
- #205 

similar to "topics".

The example of input data types I have curated for now: 
- genomic
- clinical
- imaging
- ehr
- RNA sequencing
- MRI
- survival
...
- other

### CODE REVIEW

**Review by tschaffter — COMMENTED** [review]
This PR makes me think of the [Cancer Complexity Knowledge Portal](https://staging.cancercomplexity.synapse.org/Explore/Tools).

This portal is a tool registry. For a tool registry, I like that it has:

- Input data
- Output data
- Input format
- Output format

Do you think that the user would appreciate this granularity in the challenge registry? Would there be difficulties in adding all or only a few of these filters to the challenge registry?

**Comment by tschaffter** [comment]
Our `Submission Type` is similar to `Tool Type`. However if we accept non-tool submissions (e.g. research paper), `Submission Type` is a suitable name.

**Comment by rrchai** [comment]
> Our Submission Type is similar to Tool Type. However if we accept non-tool submissions (e.g. research paper), Submission Type is a suitable name.

`submission type` makes more sense to me. `Tool Type` sounds more like what tools used in the challenges and this could be overlapped with programming languages/skills to me.

**Comment by rrchai** [comment]
> Do you think that the user would appreciate this granularity in the challenge registry? 

I cannot speak for everyone, but personally for `Output data`,  `Input format` and `Output format`, I feel they are good to have, but not necessary for the use case on the challenges data.

**Comment by rrchai** [comment]
@tschaffter in terms of the filter name,  my preference: "Input Data" > "Data Type" > "Training Data"

File: openapi/components/schemas/ChallengeCreateRequest.yaml [hunk 1]
```diff
@@ -34,6 +34,12 @@ properties:
       $ref: Topic.yaml
     # uniqueItems: true
     maxItems: 30
+  inputDataTypes:
+    type: array
+    items:
+      $ref: inputDataType.yaml
```
Comment (rrchai) [inline openapi/components/schemas/ChallengeCreateRequest.yaml:40]: @tschaffter based on our discussion on Monday, I have change `dataTypes` to `inputDataTypes`

**Review by tschaffter — APPROVED** [review]
LGTM!

**Pull request merged**